### PR TITLE
Added tests for getType and null varargs in expression language 5.0

### LINKED ISSUE #20456: Expression Language 5.0: Explicitly document that ELResolver.getType() must return null if either the ELResolver or the resolved property is read-only. This changes the documented behaviour of the StaticFieldELResolver.
Spec issue: `https://github.com/jakartaee/expression-language/issues/168`

See "A.1. Changes between 5.0 and 4.0" here: https://github.com/jakartaee/expression-language/blob/master/spec/src/main/asciidoc/ELSpec.adoc

### LINKED ISSUE #20460: Expression Language 5.0: Avoid a NullPointerException when passing null for varargs.
Spec issue: `https://github.com/jakartaee/expression-language/issues/183`

See "A.1. Changes between 5.0 and 4.0" here: https://github.com/jakartaee/expression-language/blob/master/spec/src/main/asciidoc/ELSpec.adoc

### PR DESCRIPTION
**Pull request by isaacrivriv**
fixes #20460 
fixes #20456

### CODE REVIEW

**Comment by isaacrivriv** [comment]
#build

File: dev/com.ibm.ws.el.3.0_fat/test-applications/TestVarargsMatching.war/src/com/ibm/ws/el30/fat/varargstest/EL30VarargsMethodMatchingServlet.java [hunk 1]
```diff
@@ -116,6 +116,16 @@ public void testString_VarargsBird() throws Exception {
 
     }
 
+    @Test
+    public void selectMethodWithNullVarargs() throws Exception {
```
Comment (pnicolucci) [inline dev/com.ibm.ws.el.3.0_fat/test-applications/TestVarargsMatching.war/src/com/ibm/ws/el30/fat/varargstest/EL30VarargsMethodMatchingServlet.java:120]: @isaacrivriv  you're adding this to the el.3.0 FAT b/c our 3.0 implementation was already behaving this way?

File: dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java [hunk 2]
```diff
@@ -33,6 +38,7 @@
  * Servlet for testing basic changes in EL 5.0
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java]: Can we update this to spell out "Expression Language"?

File: dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java [hunk 3]
```diff
@@ -95,19 +100,36 @@ public void testGenerics() throws Exception {
...
-    public class SimpleBean {
-
-        public int getNumber() {
-            return 25;
-        }
-
+    /**
+     *
+     * Expression Language 5.0 in Jakarta EE10 clarified and documented getType must return null if property is read-only
+     * and switched implementation of StaticFieldELResolver getType to match this behavior as it was inconsistent before EL 5.0.
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java]: Update to spell out "Expression Language"

File: dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java [hunk 4]
```diff
@@ -95,19 +100,36 @@ public void testGenerics() throws Exception {
...
+    public void testGetType_returnsNull() throws Exception {
+        ELManager elm = new ELManager();
+        StaticFieldELResolver resolver = new StaticFieldELResolver();
+        elm.addELResolver(resolver);
+        ELContext context = elm.getELContext();
+        context.setPropertyResolved(false);
+        Object bean = new ELClass(SimpleBean.class);
+        Class<?> type = resolver.getType(context, bean, "staticString");
+        assertTrue("When calling getType for StaticFieldELResolver, the property was not resolved!", context.isPropertyResolved());
+        assertNull("When calling getType for StaticFieldELResolver, did not return null instead returned: " +type, type);
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java]: "When calling getType for StaticFieldELResolver, the returned value was not null and instead was:" also add a space after `+`

File: dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java [hunk 5]
```diff
@@ -95,19 +100,36 @@ public void testGenerics() throws Exception {
...
+
+        BeanELResolver  beanResolver = new BeanELResolver();
+        elm.addELResolver(beanResolver);
+        context.setPropertyResolved(false);
+        type = beanResolver.getType(context, new SimpleBean(), "simpleProperty");
+        assertTrue("When calling getType for BeanELResolver, the property was not resolved!", context.isPropertyResolved());
+        assertNull("When calling getType for BeanELResolver, did not return null instead returned: " +type, type);
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java]: Same comment as above for StaticFieldELResolver

File: dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/SimpleBean.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.el50.fat.basic.servlets;
+
+import java.util.Arrays;
+
+/**
+ * Bean for testing basic changes in EL 5.0
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/SimpleBean.java]: Spell out Expression Language.

File: dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java (hunk 2 above)
Comment (isaacrivriv) [reply dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java]: Sure, I'll go through a couple and update them all as I go

File: dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java (hunk 4 above)
Comment (isaacrivriv) [reply dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java]: I'll work on it

File: dev/com.ibm.ws.el.3.0_fat/test-applications/TestVarargsMatching.war/src/com/ibm/ws/el30/fat/varargstest/EL30VarargsMethodMatchingServlet.java (hunk 1 above)
Comment (isaacrivriv) [reply dev/com.ibm.ws.el.3.0_fat/test-applications/TestVarargsMatching.war/src/com/ibm/ws/el30/fat/varargstest/EL30VarargsMethodMatchingServlet.java:120]: Technically, the issue is specific to Eclipse's EL API and Tomcat already had guards in place to protect against these error. I'll update the issue with my findings for this but discussed with @volosied and decided to add an additional test for verifying null varargs and this was the best place to include it in.

Comment (pnicolucci) [reply dev/com.ibm.ws.el.3.0_fat/test-applications/TestVarargsMatching.war/src/com/ibm/ws/el30/fat/varargstest/EL30VarargsMethodMatchingServlet.java:120]: Ok, thanks for the clarification. This all makes sense.

**Review by pnicolucci — APPROVED** [review]

**Comment by isaacrivriv** [comment]
#build

**Comment by isaacrivriv** [comment]
#libby
